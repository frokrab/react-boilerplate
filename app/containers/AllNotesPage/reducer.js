import { fromJS } from 'immutable';

import { GET_NOTES, GET_NOTES_SUCCESS, GET_NOTES_FAILURE } from './constants';

export const initialState = fromJS({
  notes: [],
  loading: false,
  error: false,
});

const allNotesReducer = (state = initialState, action) => {
  if (action.type === GET_NOTES) {
    return state.set('loading', true).set('error', false);
  }
  if (action.type === GET_NOTES_SUCCESS) {
    return state.set('loading', false).set('notes', action.notes);
  }
  if (action.type === GET_NOTES_FAILURE) {
    return state.set('loading', false).set('error', action.error);
  }
  return state;
};

export default allNotesReducer;
