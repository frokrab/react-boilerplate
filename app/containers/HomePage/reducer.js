import { fromJS } from 'immutable';

import { CHANGE_INPUT } from './constants';

export const initialState = fromJS({
  input: '',
});

const homeReducer = (state = initialState, action) => {
  if (action.type === CHANGE_INPUT) {
    return state.set('input', action.input);
  }
  return state;
};

export default homeReducer;
