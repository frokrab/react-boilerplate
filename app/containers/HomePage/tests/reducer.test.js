import reducer from '../reducer';
import {
  changeInput,
  saveNote,
  saveNoteSuccess,
  saveNoteFailure,
} from '../actions';

describe('HomePage reducer', () => {
  it('returns initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });
  it('handles changeInput action', () => {
    const input = 'testing';
    expect(reducer(undefined, changeInput(input))).toMatchSnapshot();
  });
  it('handles saveNote action', () => {
    expect(reducer(undefined, saveNote())).toMatchSnapshot();
  });
  it('handles saveNoteSuccess action', () => {
    expect(reducer(undefined, saveNoteSuccess())).toMatchSnapshot();
  });
  it('handles saveNoteFailure action', () => {
    const error = 'error';
    expect(reducer(undefined, saveNoteFailure(error))).toMatchSnapshot();
  });
});
