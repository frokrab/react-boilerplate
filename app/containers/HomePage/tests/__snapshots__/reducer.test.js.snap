// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage reducer handles changeInput action 1`] = `
Immutable.Map {
  "input": "testing",
  "loading": false,
  "error": false,
}
`;

exports[`HomePage reducer handles saveNote action 1`] = `
Immutable.Map {
  "input": "",
  "loading": true,
  "error": false,
}
`;

exports[`HomePage reducer handles saveNoteFailure action 1`] = `
Immutable.Map {
  "input": "",
  "loading": false,
  "error": "error",
}
`;

exports[`HomePage reducer handles saveNoteSuccess action 1`] = `
Immutable.Map {
  "input": "",
  "loading": false,
  "error": false,
}
`;

exports[`HomePage reducer returns initial state 1`] = `
Immutable.Map {
  "input": "",
  "loading": false,
  "error": false,
}
`;
